crates/wasmtime/src/runtime/vm/component/libcalls.rs

```diff
@@ -1267,3 +1267,15 @@ unsafe fn error_context_drop(
             )
     })
 }
+
+#[cfg(feature = "threads")]
+unsafe fn thread_spawn_indirect(
+    vmctx: NonNull<VMComponentContext>,
+    table: u32,
+    element: u32,
+    context: u32,
+) -> Result<u32> {
+    ComponentInstance::from_vmctx(vmctx, |instance| {
+        instance.thread_spawn_indirect(table, element, context)
```

Comment:
It's unclear to me which parts should live up at this level and which should live in `ComponentInstance::thread_spawn_indirect`. Any suggestions?

Reply:
Stylistic an up-to-you,  but generally "simple things" and "ABI things" in this file and meat elsewhere. So e.g. converting `u32` to more typed things here, but leaving the meat of the implementation elsewhere is fine.